external/script/trials.lua

```diff
@@ -0,0 +1,906 @@
+						start.trials.combocounter = 0
+					end
+				end
+				-- Next, if microstep is exceeded, go to next trial step
+				if start.trials.currenttrialmicrostep > start.trials.trial[ct].trialstep[cts].numofmicrosteps then
+					start.trials.currenttrialmicrostep = 1
+					start.trials.currenttrialstep = cts + 1
+					start.trials.combocounter = combocount()
+					start.trials.pauseuntilnexthit = start.trials.trial[ct].trialstep[cts].validuntilnexthit[ctms]
+					if start.trials.currenttrialstep > #start.trials.trial[ct].trialstep then
+						-- If trial step was last, go to next trial and display success banner
+						if start.trials.trialadvancement then
+							start.trials.currenttrial = ct + 1
+						end
+						start.trials.currenttrialstep = 1
+						start.trials.combocounter = 0
+						if ct < #start.trials.trial or (not start.trials.trialadvancement and ct == #start.trials.trial) then
+							if (motif.trials_mode.success_front_displaytime == -1) and (motif.trials_mode.success_bg_displaytime == -1) then
+								start.trials.draw.success = math.max(animGetLength(motif.trials_mode.success_front_data), animGetLength(motif.trials_mode.success_bg_data), motif.trials_mode.success_text_displaytime)
+							else
+								start.trials.draw.success = math.max(motif.trials_mode.success_front_displaytime, motif.trials_mode.success_bg_displaytime, motif.trials_mode.success_text_displaytime)
+							end
+							if motif.trials_mode.resetonsuccess == "true" then
+								start.trials.draw.fadein = motif.trials_mode.fadein_time
+								start.trials.draw.fadeout = motif.trials_mode.fadeout_time
+								start.trials.draw.fade = start.trials.draw.fadein + start.trials.draw.fadeout
+							end
+						end
+					end
+				end
+			end
+		elseif ((combocount() == 0 and not start.trials.trial[ct].trialstep[cts].isnohit[ctms]) and not start.trials.pauseuntilnexthit) or (start.trials.pauseuntilnexthit and combocount() > start.trials.combocounter) then
+			start.trials.currenttrialstep = 1
+			start.trials.currenttrialmicrostep = 1
+			start.trials.combocounter = 0
+			start.trials.trial[ct].trialstep[cts].stephitscount[ctms] = 0
+			start.trials.trial[ct].trialstep[cts].combocountonstep[ctms] = 0
+			start.trials.pauseuntilnexthit = false
+		end
+	end
+	--If the trial was completed successfully, draw the trials success
+	if start.trials.draw.success > 0 then
+		f_trialsSuccess('success', ct)
+	elseif start.trials.draw.fade > 0 and motif.trials_mode.resetonsuccess == "true" then
+		if start.trials.draw.fade < start.trials.draw.fadein + start.trials.draw.fadeout then
+			f_trialsFade()
+		else
+			player(2)
+			if stateno() == 0 then
+				f_trialsFade()
+			end
+			player(1)
+		end
+	end
+end
+
+-- Find trials files and parse them; append t_selChars table
+function trials.f_parseTrials(row)
+	i = 0 --Trial number
+	j = 0 --TrialStep number
+	trial = {}
+	local trialsFile = main.f_fileRead(main.t_selChars[row].trialspath, "r")
```

Comment:
also I tested this build and your sample of KFM trials.def but panicError caused on this f_fileRead even there IS trials.def...
```suggestion
	local trialsFile = main.f_fileRead(main.t_selChars[row].dir .. main.t_selChars[row].trialspath, "r")
```

Reply:
Thanks, it resolved the path in MacOS, but you're right, what I had wasn't working in x64. I accepted the change, try it again @bluish-elvis

Reply:
This change was breaking for MacOS but I included a change that leveraged your first suggestion as well.

Reply:
Confirmed it's work. thanks!